src/linter.js

```diff
@@ -154,104 +152,43 @@ export default class Linter {
-            },
-            maxWidth: (maxColumns - 35) * 0.25,
-          },
-          description: {
-            headingTransform: () => {
-              return i18n._('Description');
-            },
-            maxWidth: (maxColumns - 35) * 0.5,
-          },
-          file: {
-            headingTransform: () => {
-              return i18n._('File');
-            },
-            maxWidth: (maxColumns - 35) * 0.25,
-          },
-          line: {
-            headingTransform: () => {
-              return i18n._('Line');
-            },
-            maxWidth: 6,
-          },
-          column: {
-            headingTransform: () => {
-              return i18n._('Column');
-            },
-            maxWidth: 6,
-          },
-        };
-
-        const outputColumns = [
-          'code',
-          'message',
-          'description',
-          'file',
-          'line',
-          'column',
-        ];
-
-        // If the terminal is this small we cave and don't size things
-        // contextually anymore.
-        if (maxColumns < 60) {
-          delete outputColumns[outputColumns.indexOf('column')];
-          delete outputConfig.column;
-          delete outputColumns[outputColumns.indexOf('description')];
-          delete outputConfig.description;
-          delete outputColumns[outputColumns.indexOf('line')];
-          delete outputConfig.line;
-
-          outputConfig.message.maxWidth = 15;
-          outputConfig.file.maxWidth = 15;
-        } else if (maxColumns < 78) {
-          delete outputColumns[outputColumns.indexOf('description')];
-          delete outputConfig.description;
-
-          outputConfig.message.maxWidth = (maxColumns - 47) * 0.5;
-          outputConfig.file.maxWidth = (maxColumns - 35) * 0.5;
-        }
-
-        out.push(`${messageType.toUpperCase()}:`);
+        out.push(this.colorize(type)(`${messageType.toUpperCase()}:`));
         out.push('');
-        out.push(
-          columnify(this.output[messageType], {
-            maxWidth: 35,
-            columns: outputColumns,
-            columnSplitter: '   ',
-            config: outputConfig,
-          })
-        );
+
+        const groupedMessages = {};
+        this.output[messageType].forEach((message) => {
+          if (!groupedMessages[message.code]) {
+            groupedMessages[message.code] = [];
+          }
+          groupedMessages[message.code].push(message);
+        });
+
+        Object.entries(groupedMessages).forEach(([code, list]) => {
+          out.push(this.colorize(type)(`  ${code}`));
+          out.push(`  ${list[0].message}`);
+          list.forEach((message) => {
+            const location = message.file
+              ? `${message.file}${message.line ? `:${message.line}` : ''}${
+                  message.column ? `:${message.column}` : ''
+                }`
+              : 'N/A';
+            out.push(`    ${location}`);
+          });
+          out.push('');
+        });
```

Comment:
@willdurand @rpl This code is straight out of Claude. We can bikeshed the output but I think it's a vast improvement already.

Codebase cleanup and tests update will follow after approval

Reply:
It would have been good to discuss in https://github.com/mozilla/addons-linter/issues/5423 first. I don't really have a strong opinion on the current linter output but - at the same time - this works relatively well and that isn't necessarily an area where I'd change code, but that's my personal opinion on this.

Reply:
I opened a PR right after the issue in order to demo the output and changes required. It's always easier to discuss things when there's an exact desired output, especially in a project that evolves very slowly.